frontend/src/Components/modals/EditUserModal.tsx

```diff
@@ -0,0 +1,33 @@
+import { forwardRef } from 'react';
+import { CRUDModalProps, FormModal, userInputs } from '.';
+import { FieldValues, SubmitHandler } from 'react-hook-form';
+import { User } from '@/common';
+import API from '@/api/api';
+import { useCheckResponse } from '@/Hooks/useCheckResponse';
+
+export const EditUserModal = forwardRef(function (
+    { mutate, target }: CRUDModalProps<User>,
+    editUserModal: React.ForwardedRef<HTMLDialogElement>
+) {
+    const checkResponse = useCheckResponse({
+        mutate: mutate,
+        refModal: editUserModal
+    });
+    const editUser: SubmitHandler<FieldValues> = async (data) => {
+        const resp = await API.patch(`users/${target?.id}`, data);
+        checkResponse(
+            resp.success,
+            'Unable to modify user',
+            'User updated successfully'
+        );
```

Comment:
Checking to make sure we can't add similar message logic like we do in add user because the backend doesn't give a detailed enough error message if the edit user fails. This is what I get when I try to update a user to take an existing user name.

```
{"message":"username","data":null}
```